fault.c

```diff
@@ -157,10 +160,161 @@ static int kibosh_fault_read_delay_check(struct kibosh_fault_read_delay *fault,
     if (strncmp(path, fault->prefix, strlen(fault->prefix)) != 0) {
         return 0;
     }
-    milli_sleep(fault->delay_ms);
+    // apply fraction
+    srand((int) round(time(0)*RAND_FRAC));
+    if (RAND_FRAC <= fault->fraction)
+        milli_sleep(fault->delay_ms);
     return 0;
 }
 
+/**
+ * kibosh_fault_unwritable
```

Comment:
please add more description to the functions, including any parameters and return values. Similar for all functions added.

Reply:
Descriptions to the functions are included in the header file.